### ISSUE TITLE
ACV

### ISSUE THREAD

**Issue by Niyasmkkv** [iptv-org/iptv#4151]
**_Channel Name:_** ACV
**_Channel ID:_** ACV.in
**_Broken Link (from playlist):_** https://acv.asianetmobiletvplus.com/webstreams/8f8e72769cb3e3a6e27c220e1e3887b8.smil/playlist.m3u8
**_Possible Replacement (optional):_** xxx
**_Notes (optional):_** The website is down for the acv group of channels. But their app working fine with all these channels.

**Comment by Niyasmkkv** [comment]
@Dum4G @LaneSh4d0w Found this link from the app. Check this working for Rosebowl
http://cdn.asianetmobiletvplus.com/channels/rosebowl_7eb4dc1f3240c8eb776d41b95bd1d197/chunks.m3u8?nimblesessionid=24180448

**Comment by Dum4G** [comment]
nimblesession part isn't needed, nice find

**Comment by Niyasmkkv** [comment]
I couldn't find the other links though.

**Comment by Niyasmkkv** [comment]
These links are not playing on kodi.

**Comment by Niyasmkkv** [comment]
@Sphinxroot @LaneSh4d0w  can you guys try changing the link in nay way so that it will be compatible with Kodi?

**Comment by Sphinxroot** [comment]
in kodi (Playlistloader) it  depend where the stream come from   , it could take  referrer  only or user-agent  or both  to get it working 

|Referer=  {link to referrer}
|User-Agent=  {Agent} 
vlc
stream
Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148
or other  it will need to be tested 1 by 1 to find out

https:// serverstream.link .m3u8 |Referer=URL|User-Agent=

some Website have  { strick origin } , those are hard to make them working out of the web site , exepted if somewhere , there a re-stream server of the source.

but for me the link http://cdn.asianetmobiletvplus.com/channels/rosebowl_7eb4dc1f3240c8eb776d41b95bd1d197/chunks.m3u8  is 403 so could be GEO stream

**Comment by Niyasmkkv** [comment]
Tried IPTV pro app and kodi, it's not working. Only working with vlc. Is it the hls format or something?

**Comment by Sphinxroot** [comment]
how strange this is lol

#EXTINF:1 ,ACV Medley
https://acv.asianetmobiletvplus.com/webstreams/95ef9473f8e0c8898d58b16300dbb7b5/playlist.m3u8|User-Agent=Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148

#EXTINF:1 ,Rosebowl
http://cdn.asianetmobiletvplus.com/channels/rosebowl_7eb4dc1f3240c8eb776d41b95bd1d197/chunks.m3u8|User-Agent=stream

**Comment by LaneSh4d0w** [comment]
> how strange this is lol
> 
> #EXTINF:1 ,ACV Medley
> https://acv.asianetmobiletvplus.com/webstreams/95ef9473f8e0c8898d58b16300dbb7b5/playlist.m3u8|User-Agent=Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148
> 
> #EXTINF:1 ,Rosebowl
> http://cdn.asianetmobiletvplus.com/channels/rosebowl_7eb4dc1f3240c8eb776d41b95bd1d197/chunks.m3u8|User-Agent=stream
> 

Between the cdn and the acv server, they seem to accept specific user-agents haha

**Comment by Sphinxroot** [comment]
well all other acv link dont work  .. mediamanifest absent if i change link to cdn  it the same 403 so might be geo now

**Comment by Niyasmkkv** [comment]
> well all other acv link dont work .. mediamanifest absent if i change link to cdn it the same 403 so might be geo now

Share the link with me, Let me try.

**Comment by Sphinxroot** [comment]
#EXTINF:-1 tvg-logo="https://i.imgur.com/UBZg9Yl.png" group-title="INDIA",ACV (Opt-1) (New!)
https://acv.asianetmobiletvplus.com/channels/8f8e72769cb3e3a6e27c220e1e3887b8.smil/index.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/UBZg9Yl.png" group-title="INDIA",ACV (Opt-2) (New!)
https://acv.asianetmobiletvplus.com/webstreams/8f8e72769cb3e3a6e27c220e1e3887b8.smil/playlist.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/AllTF2K.png" group-title="INDIA",ACV Jukebox (Opt-1) (New!)
https://acv.asianetmobiletvplus.com/channels/acvjukebox_awannbgiynqynhufohawnvbmlgglfpuc/index.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/AllTF2K.png" group-title="INDIA",ACV Jukebox (Opt-2) (New!)
https://acv.asianetmobiletvplus.com/webstreams/acvjukebox_awannbgiynqynhufohawnvbmlgglfpuc/playlist.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/UjqTQgr.png" group-title="INDIA",ACV News (New!)
https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/QnDmR8W.png" group-title="INDIA",ACV Utsav (Opt-1) (New!)
https://acv.asianetmobiletvplus.com/webstreams/021c9292219a98f899a7b74f0f34baa7.smil/playlist.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/QnDmR8W.png" group-title="INDIA",ACV Utsav (Opt-2) (New!)
https://acv.asianetmobiletvplus.com/channels/021c9292219a98f899a7b74f0f34baa7.smil/index.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/Hcp2Noi.png" group-title="INDIA",ACV Utsav Plus (Opt-1) (New!)
https://acv.asianetmobiletvplus.com/channels/7eb4dc1f3240c8eb776d41b95bd1d197.smil/index.m3u8
 
#EXTINF:-1 tvg-logo="https://i.imgur.com/Hcp2Noi.png" group-title="INDIA",ACV Utsav Plus (Opt-2) (New!)
https://acv.asianetmobiletvplus.com/webstreams/7eb4dc1f3240c8eb776d41b95bd1d197.smil/playlist.m3u8

 surly the same we use to have

**Comment by LaneSh4d0w** [comment]
Since we're talking Asianet : 
#EXTINF:-1 tvg-logo="" group-title="Movies",Asianet Movies [Geo-blocked]
http://asianetmovies-i.akamaihd.net/hls/live/569919/asianetmovies/master.m3u8
#EXTINF:-1 tvg-logo="" group-title="Movies",Asianet HD [Geo-blocked]
http://asianethd-i.akamaihd.net/hls/live/569828/asianethd/master.m3u8

**Comment by LaneSh4d0w** [comment]
Streams answer but 404s on chunklist retrieval, using streamlink

**Comment by Niyasmkkv** [comment]
Not working on vlc for me.

**Comment by Niyasmkkv** [comment]
> #EXTINF:-1 tvg-logo="https://i.imgur.com/UBZg9Yl.png" group-title="INDIA",ACV (Opt-1) (New!)
> https://acv.asianetmobiletvplus.com/channels/8f8e72769cb3e3a6e27c220e1e3887b8.smil/index.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/UBZg9Yl.png" group-title="INDIA",ACV (Opt-2) (New!)
> https://acv.asianetmobiletvplus.com/webstreams/8f8e72769cb3e3a6e27c220e1e3887b8.smil/playlist.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/AllTF2K.png" group-title="INDIA",ACV Jukebox (Opt-1) (New!)
> https://acv.asianetmobiletvplus.com/channels/acvjukebox_awannbgiynqynhufohawnvbmlgglfpuc/index.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/AllTF2K.png" group-title="INDIA",ACV Jukebox (Opt-2) (New!)
> https://acv.asianetmobiletvplus.com/webstreams/acvjukebox_awannbgiynqynhufohawnvbmlgglfpuc/playlist.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/UjqTQgr.png" group-title="INDIA",ACV News (New!)
> https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/QnDmR8W.png" group-title="INDIA",ACV Utsav (Opt-1) (New!)
> https://acv.asianetmobiletvplus.com/webstreams/021c9292219a98f899a7b74f0f34baa7.smil/playlist.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/QnDmR8W.png" group-title="INDIA",ACV Utsav (Opt-2) (New!)
> https://acv.asianetmobiletvplus.com/channels/021c9292219a98f899a7b74f0f34baa7.smil/index.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/Hcp2Noi.png" group-title="INDIA",ACV Utsav Plus (Opt-1) (New!)
> https://acv.asianetmobiletvplus.com/channels/7eb4dc1f3240c8eb776d41b95bd1d197.smil/index.m3u8
> 
> #EXTINF:-1 tvg-logo="https://i.imgur.com/Hcp2Noi.png" group-title="INDIA",ACV Utsav Plus (Opt-2) (New!)
> https://acv.asianetmobiletvplus.com/webstreams/7eb4dc1f3240c8eb776d41b95bd1d197.smil/playlist.m3u8
> 
> surly the same we use to have
> 
> same link can be found here https://mhdtvworld.xyz/channel/malayalam/

None of these are working.

**Comment by Niyasmkkv** [comment]
@LaneSh4d0w can you convert this file again, I wanna see whether it will work in kodi or not. The current one is not working.
http://cdn.asianetmobiletvplus.com/channels/adngold_dibbcspwxywdcuwawgrvurjwitwbiksl/chunks.m3u8

**Comment by LaneSh4d0w** [comment]
> 
> 
> @LaneSh4d0w can you convert this file again, I wanna see whether it will work in kodi or not. The current one is not working.
> http://cdn.asianetmobiletvplus.com/channels/adngold_dibbcspwxywdcuwawgrvurjwitwbiksl/chunks.m3u8

Works for me if user agent is "stream".

**Comment by Niyasmkkv** [comment]
It's working on kodi with iptv client for me,  Working on VLC. Also, when I try to download the m3u file by adding the URL to browser tab, it shows forbidden.

**Comment by LaneSh4d0w** [comment]
> 
> 
> It's working on kodi with iptv client for me, Working on VLC. Also, when I try to download the m3u file by adding the URL to browser tab, it shows forbidden.

Because the browser will print its user-agent, and it seems that it doesn't like it. If you make a request from the browser with User-Agent "stream", it'll work.

**Comment by Niyasmkkv** [comment]
Correction. not working on kodi with iptv client for me.

**Comment by LaneSh4d0w** [comment]
Works the same with Flowers

![image](https://user-images.githubusercontent.com/30985701/130109734-378099dc-c587-4b70-be6d-563b5ed85edf.png)

**Comment by Niyasmkkv** [comment]
I am using Kodi for the IPTV, so I am not able to use these asianet channels. Even the old asianet server liks were not paying on Kodi.

**Comment by LaneSh4d0w** [comment]
@Sphinxroot is the one that knows how to deal with Kodi. Maybe he knows about User-agent and stuff.

**Comment by LaneSh4d0w** [comment]
https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8

Works on browser, but 404 on chunklist retrieval.

**Comment by Niyasmkkv** [comment]
Still not playing on Kodi.

**Comment by LaneSh4d0w** [comment]
```[cli][info] Found matching plugin hls for URL https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8
[cli][info] Available streams: 1000k (worst, best)
[cli][info] Opening stream: 1000k (hls)
[cli][error] Try 1/1: Could not open stream <HLSStream('https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/chunklist_b996000.m3u8?nimblesessionid=24363460', 'https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8')> (Could not open stream: Unable to open URL: https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/chunklist_b996000.m3u8?nimblesessionid=24363460 (404 Client Error: Not Found for url: https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/chunklist_b996000.m3u8?nimblesessionid=24363460))
error: Could not open stream <HLSStream('https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/chunklist_b996000.m3u8?nimblesessionid=24363460', 'https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8')>, tried 1 times, exiting```

**Comment by Niyasmkkv** [comment]
> https://acv.asianetmobiletvplus.com/webstreams/3e85eb4c12bd2110d3f495676205d50a.smil/playlist.m3u8
> 
> Works on browser, but 404 on chunklist retrieval.

This link is not working. only cdn servers are working.

**Comment by Niyasmkkv** [comment]
@Sphinxroot  do i jave to add any extra add on if some channels are not playing on kodi with language playlist from IPTV simple client? Except acv channels, all channels are working fine.

**Comment by Sphinxroot** [comment]
> @Sphinxroot do i jave to add any extra add on if some channels are not playing on kodi with language playlist from IPTV simple client? Except acv channels, all channels are working fine.

try with playlistloader addon  for kodi 18.9 or matrix i got them on my git be sure to have  inputstream adaptative  , rtmp input , inputstream helper , youtube-dl control ,  and i use streamlink in kodi because you can script channel in the folder for yourself
you can also play plugin and script that have play command in it


i use pvr iptv simple client only with a list that have EPG built in

**Comment by Dum4G** [comment]
I got everything working except ACV. Is it the only channel to fix now?

**Comment by LaneSh4d0w** [comment]
Yup !

**Comment by Niyasmkkv** [comment]
Still no update on this? I can find this channel on their mobile tv app and android tv app. But not able to extract the links.
@LaneSh4d0w @Dum4G

**Comment by LaneSh4d0w** [comment]
Use some SSL pinning with Frida and try to see if you can get it

**Comment by LaneSh4d0w** [comment]
I personally cannot access the Asianet app from where I am, so either you know how to export your app as an apk file, either I shall need to dig more on it.

**Comment by LaneSh4d0w** [comment]
![image](https://user-images.githubusercontent.com/30985701/136543899-572f7594-4846-4a18-a4fc-da32bbbd16d5.png)

Cheap looking application, is it like that ?

**Comment by freearhey** [comment]
Closed in favor of #12277
